tests/test_approximate.py

```diff
@@ -0,0 +1,489 @@
+# -*- coding: utf-8 -*-
+
+import numpy as np
+from mabwiser.mab import LearningPolicy, NeighborhoodPolicy
+from mabwiser.approximate import _LSHNearest
+from tests.test_base import BaseTest
+
+
+class ApproximateTest(BaseTest):
+
+    def test_hash_function(self):
+        seed = 11
+        n_dimensions = 5
+        n_tables = 5
+        table_to_plane = {i: [] for i in range(n_tables)}
+        rng = np.random.RandomState(seed)
+        contexts = np.array([[rng.rand() for _ in range(7)] for _ in range(10)])
+
+        table_to_plane = {i: rng.standard_normal(size=(contexts.shape[1],
+                                                       n_dimensions))
+                          for i in table_to_plane.keys()}
+
+        hash_lists = []
+        for k in table_to_plane.keys():
+            hashes = _LSHNearest.get_context_hash(contexts, table_to_plane[k])
+            hash_lists.append(list(hashes))
+        self.assertListEqual(hash_lists[0], [22, 14, 11, 14, 18,  9,  6,  7, 14, 11])
+        self.assertListEqual(hash_lists[1], [28, 29, 21, 7, 20, 28,  4, 28, 28, 20])
+        self.assertListEqual(hash_lists[2], [3, 19, 19, 18, 19, 19, 19, 19, 19, 19])
+        self.assertListEqual(hash_lists[3], [1, 1, 9, 16, 1, 11, 1, 7, 3, 1])
+        self.assertListEqual(hash_lists[4], [18, 16, 20, 26, 18, 19, 20, 16, 16, 16])
+
+    def test_greedy0_d2(self):
+
+        arms, mab = self.predict(arms=[1, 2, 3, 4],
+                                 decisions=[1, 1, 1, 2, 2, 3, 3, 3, 3, 3],
+                                 rewards=[0, 1, 1, 0, 0, 0, 0, 1, 1, 1],
+                                 learning_policy=LearningPolicy.EpsilonGreedy(epsilon=0),
+                                 neighborhood_policy=NeighborhoodPolicy.LSHNearest(n_dimensions=2),
+                                 context_history=[[0, 1, 2, 3, 5], [1, 1, 1, 1, 1], [0, 0, 1, 0, 0],
+                                                  [0, 2, 2, 3, 5], [1, 3, 1, 1, 1], [0, 0, 0, 0, 0],
+                                                  [0, 1, 4, 3, 5], [0, 1, 2, 4, 5], [1, 2, 1, 1, 3],
+                                                  [0, 2, 1, 0, 0]],
+                                 contexts=[[0, 1, 2, 3, 5], [1, 1, 1, 1, 1]],
+                                 seed=123456,
+                                 num_run=1,
+                                 is_predict=True)
+
+        self.assertListEqual(arms, [3, 1])
+
+    def test_greedy0_d2_single_test(self):
+
+        arms, mab = self.predict(arms=[1, 2, 3, 4],
+                                 decisions=[1, 1, 1, 2, 2, 3, 3, 3, 3, 3],
+                                 rewards=[0, 1, 1, 0, 0, 0, 0, 1, 1, 1],
+                                 learning_policy=LearningPolicy.EpsilonGreedy(epsilon=0),
+                                 neighborhood_policy=NeighborhoodPolicy.Radius(2),
```

Comment:
why radious tested in test_approx?.

Reply:
Whoops! Apparently PyCharm's search functionality is case sensitive